pkg/machinery/client/resolver/roundrobin.go

```diff
@@ -58,13 +60,30 @@ type roundRobinResolver struct {
 	port   int
 }
 
+func EnsureEndpointsHavePorts(endpoints []string, defaultPort int) ([]string, error) {
+	for i, endpoint := range endpoints {
+		if !sideronet.AddressContainsPort(endpoint) {
```

Comment:
like I put in my previous comment, let's use `net.SplitHostPort` to attempt to split a port. if it succeeds, we don't need to do anything, otherwise we do `JoinHostPort`

Reply:
I saw your comment and tried it, but `net.SplitHostPort` returns an error if there is no port and this function can also return other errors, so the only way to distinguish them would be a weird string comparison.

Reply:
no big deal, if it returns an error, there's no port, and that's all we need to know. we don't need an error here

Reply:
done